jlc_lib/generate_bom.py

```diff
@@ -54,45 +54,50 @@ def GenerateBOM(input_filename, output_filename, opts):
         field_value = c.getField(field_name)
 
         if LCSC_PART_NUMBER_MATCHER.match(field_value):
-          lcsc_part_numbers.append(field_value)
+          lcsc_part_numbers.add(field_value)
         else:
-          lcsc_part_numbers.append(None)
+          lcsc_part_numbers.add(None)
```

Comment:
Just do the None check here, so we don't have to deal with Nones in the set later.
```
elif not opts.assume_same_lcsc_partnumber:
  logging.error(...)
  return False
```
Same for footprint:
```
if c.getFootprint() is None or c.getFootprint() == "":
  logging.error(...)
  return False
else:
  footprints.add(c.getFootprint())
```

Reply:
For the part numbers this was my first approach, but then this will already fail before passing all components of that group.

Reply:
Is that a problem if we are going to return False later anyways?

Reply:
I wanted to show the error of possible different part numbers in that case nevertheless and also a complete list of all references in that group. I could have assembled this in the error case additionally, but then would have this duplicated.

Reply:
In that case, I would prefer checking for None right after the loop, and remove it from the set. That way we still won't have to deal with Nones in the set later, and can have a full error message.

I'd like to keep the code below as simple and readable as possible, because this is likely the most bug-prone part of the whole program.

Reply:
Removed the none from the sets now entirely and store the information if nothing was sound in a separate boolean. Hope this makes it more clear.